### PR TITLE
Don't get the text content for an annotation when /NeedAppearances is true (bug 1844583)

### PR DESCRIPTION
**Pull request by calixteman**
When the flag is set, the appearance has to be generated from the value so it's useless/meaningless to extract the content from the existing appearance.

### CODE REVIEW

File: src/core/annotation.js [hunk 1]
```diff
@@ -3702,6 +3702,14 @@ class FreeTextAnnotation extends MarkupAnnotation {
     return !!this.appearance;
   }
 
+  async extractTextContent(evaluator, task, viewBox) {
+    if (this._needAppearances && this._contents) {
+      this.data.textContent = this._contents.str;
```
[the remark this answers is not in the document]
Comment (calixteman) [reply src/core/annotation.js]: Yep, no need to do that: NeedAppearances is only for forms.

**Review by Snuffleupagus — APPROVED** [review]
r=me, with passing tests; thank you!

**Pull request merged**